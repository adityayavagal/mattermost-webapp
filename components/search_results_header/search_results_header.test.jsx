// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import SearchResultsHeader from './search_results_header.jsx';

describe('components/SearchResultsHeader', () => {
    const baseProps = {
        children: 'Mentions',
        isExpanded: true,
        actions: {
            closeRightHandSide: jest.fn(),
            toggleRhsExpanded: jest.fn(),
        },
    };

    test('should match snapshot for expanded true state', () => {
        const wrapper = shallow(<SearchResultsHeader {...baseProps}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot for expanded false state', () => {
        const props = {
            ...baseProps,
            isExpanded: false
        };
        const wrapper = shallow(<SearchResultsHeader {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });
});