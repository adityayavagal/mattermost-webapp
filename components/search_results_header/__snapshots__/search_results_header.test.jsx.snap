// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/SearchResultsHeader should match snapshot for expanded false state 1`] = `
<div
  className="sidebar--right__header"
>
  <span
    className="sidebar--right__title"
  >
    Mentions
  </span>
  <div
    className="pull-right"
  >
    <OverlayTrigger
      defaultOverlayShown={false}
      delayShow={400}
      overlay={
        <Tooltip
          bsClass="tooltip"
          id="expandSidebarTooltip"
          placement="right"
        >
          <FormattedMessage
            defaultMessage="Expand Sidebar"
            id="rhs_header.expandSidebarTooltip"
            values={Object {}}
          />
        </Tooltip>
      }
      placement="top"
      trigger={
        Array [
          "hover",
          "focus",
        ]
      }
    >
      <button
        aria-label="Expand"
        className="sidebar--right__expand"
        onClick={[MockFunction]}
        type="button"
      >
        <FormattedMessage
          defaultMessage="Expand Sidebar Icon"
          id="rhs_header.expandSidebarTooltip.icon"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
        <FormattedMessage
          defaultMessage="Collapse Sidebar Icon"
          id="rhs_header.expandTooltip.icon"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </button>
    </OverlayTrigger>
    <OverlayTrigger
      defaultOverlayShown={false}
      delayShow={400}
      overlay={
        <Tooltip
          bsClass="tooltip"
          id="closeSidebarTooltip"
          placement="right"
        >
          <FormattedMessage
            defaultMessage="Close"
            id="rhs_header.closeSidebarTooltip"
            values={Object {}}
          />
        </Tooltip>
      }
      placement="top"
      trigger={
        Array [
          "hover",
          "focus",
        ]
      }
    >
      <button
        aria-label="Close"
        className="sidebar--right__close"
        id="searchResultsCloseButton"
        onClick={[MockFunction]}
        type="button"
      >
        <FormattedMessage
          defaultMessage="Close Sidebar Icon"
          id="rhs_header.closeTooltip.icon"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </button>
    </OverlayTrigger>
  </div>
</div>
`;

exports[`components/SearchResultsHeader should match snapshot for expanded true state 1`] = `
<div
  className="sidebar--right__header"
>
  <span
    className="sidebar--right__title"
  >
    Mentions
  </span>
  <div
    className="pull-right"
  >
    <OverlayTrigger
      defaultOverlayShown={false}
      delayShow={400}
      overlay={
        <Tooltip
          bsClass="tooltip"
          id="shrinkSidebarTooltip"
          placement="right"
        >
          <FormattedMessage
            defaultMessage="Collapse Sidebar"
            id="rhs_header.shrinkSidebarTooltip"
            values={Object {}}
          />
        </Tooltip>
      }
      placement="top"
      trigger={
        Array [
          "hover",
          "focus",
        ]
      }
    >
      <button
        aria-label="Expand"
        className="sidebar--right__expand"
        onClick={[MockFunction]}
        type="button"
      >
        <FormattedMessage
          defaultMessage="Expand Sidebar Icon"
          id="rhs_header.expandSidebarTooltip.icon"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
        <FormattedMessage
          defaultMessage="Collapse Sidebar Icon"
          id="rhs_header.expandTooltip.icon"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </button>
    </OverlayTrigger>
    <OverlayTrigger
      defaultOverlayShown={false}
      delayShow={400}
      overlay={
        <Tooltip
          bsClass="tooltip"
          id="closeSidebarTooltip"
          placement="right"
        >
          <FormattedMessage
            defaultMessage="Close"
            id="rhs_header.closeSidebarTooltip"
            values={Object {}}
          />
        </Tooltip>
      }
      placement="top"
      trigger={
        Array [
          "hover",
          "focus",
        ]
      }
    >
      <button
        aria-label="Close"
        className="sidebar--right__close"
        id="searchResultsCloseButton"
        onClick={[MockFunction]}
        type="button"
      >
        <FormattedMessage
          defaultMessage="Close Sidebar Icon"
          id="rhs_header.closeTooltip.icon"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </button>
    </OverlayTrigger>
  </div>
</div>
`;
